# vim: se ft=diff :
# HG changeset patch
# User Rob Campbell <rcampbell@mozilla.com>
# Date 2012-03-02 07:26
updated

# HG changeset patch
# Parent e89ed70101c0bd0b265cb312efecfe35997563c8
# User Rob Campbell <rcampbell@mozilla.com>
updated

diff --git a/browser/devtools/debugger/DebuggerUI.jsm b/browser/devtools/debugger/DebuggerUI.jsm
--- a/browser/devtools/debugger/DebuggerUI.jsm
+++ b/browser/devtools/debugger/DebuggerUI.jsm
@@ -24,6 +24,7 @@
  *   Dave Camp <dcamp@mozilla.com> (original author)
  *   Panos Astithas <past@mozilla.com>
  *   Victor Porof <vporof@mozilla.com>
+ *   Rob Campbell <rcampbell@mozilla.com>
  *
  * Alternatively, the contents of this file may be used under the terms of
  * either the GNU General Public License Version 2 or later (the "GPL"), or
@@ -71,7 +72,7 @@ DebuggerPane.prototype = {
 
     this._nbox = gBrowser.getNotificationBox(this._tab.linkedBrowser);
     this._splitter = gBrowser.parentNode.ownerDocument.createElement("splitter");
-    this._splitter.setAttribute("class", "hud-splitter");
+    this._splitter.classList.add("devtools-horizontal-splitter");
     this.frame = gBrowser.parentNode.ownerDocument.createElement("iframe");
     this.frame.height = DebuggerUIPreferences.height;
 
diff --git a/browser/devtools/debugger/debugger.css b/browser/devtools/debugger/debugger.css
--- a/browser/devtools/debugger/debugger.css
+++ b/browser/devtools/debugger/debugger.css
@@ -24,6 +24,7 @@
  *   Dave Camp <dcamp@mozilla.com>
  *   Victor Porof <vporof@mozilla.com>
  *   Panos Astithas <past@mozilla.com>
+ *   Rob Campbell <rcampbell@mozilla.com>
  *
  * Alternatively, the contents of this file may be used under the terms of
  * either the GNU General Public License Version 2 or later (the "GPL"), or
@@ -48,14 +49,6 @@
 }
 
 /**
- * Debugger statusbar
- */
-
-#dbg-statusbar {
-  -moz-appearance: statusbar;
-}
-
-/**
  * Stack frames
  */
 
@@ -92,7 +85,7 @@
  */
 
 #properties {
-  width: 250px;
+  width: 200px;
 }
 
 #variables {
diff --git a/browser/devtools/debugger/debugger.xul b/browser/devtools/debugger/debugger.xul
--- a/browser/devtools/debugger/debugger.xul
+++ b/browser/devtools/debugger/debugger.xul
@@ -23,6 +23,7 @@
    -   Dave Camp <dcamp@mozilla.com>
    -   Panos Astithas <past@mozilla.com>
    -   Victor Porof <vporof@mozilla.com>
+   -   Rob Campbell <rcampbell@mozilla.com>
    -
    - Alternatively, the contents of this file may be used under the terms of
    - either the GNU General Public License Version 2 or later (the "GPL"), or
@@ -40,6 +41,7 @@
 <?xml-stylesheet href="chrome://browser/skin/" type="text/css"?>
 <?xml-stylesheet href="chrome://browser/content/orion.css" type="text/css"?>
 <?xml-stylesheet href="chrome://browser/content/debugger.css" type="text/css"?>
+<?xml-stylesheet href="chrome://browser/skin/devtools/common.css" type="text/css"?>
 <?xml-stylesheet href="chrome://browser/skin/devtools/debugger.css" type="text/css"?>
 <!DOCTYPE window [
 <!ENTITY % debuggerDTD SYSTEM "chrome://browser/locale/devtools/debugger.dtd" >
@@ -51,28 +53,25 @@
     <xul:script type="text/javascript" src="debugger-view.js"/>
 
     <div id="body" class="vbox flex">
-        <xul:toolbar id="dbg-toolbar">
-            <xul:button id="close">&debuggerUI.closeButton;</xul:button>
-            <xul:button id="resume"/>
-            <xul:menulist id="scripts"/>
+    <xul:toolbar id="dbg-toolbar" class="devtools-toolbar">
+            <xul:toolbarbutton id="close" class="devtools-closebutton"/>
+            <xul:toolbarbutton id="resume" class="devtools-toolbarbutton"/>
+            <xul:menulist id="scripts"
+                          class="devtools-toolbarbutton"/>
+      <xul:hbox flex="1"/>
+      <xul:label id="status" class="devtools-toolbarlabel" align="end"/>
         </xul:toolbar>
         <div id="dbg-content" class="hbox flex">
             <div id="stack" class="vbox">
-                <div class="title unselectable">&debuggerUI.stackTitle;</div>
                 <div id="stackframes" class="vbox flex"></div>
             </div>
             <div id="script" class="vbox flex">
-                <div class="title unselectable">&debuggerUI.scriptTitle;</div>
                 <div id="editor" class="vbox flex"></div>
             </div>
             <div id="properties" class="vbox">
-                <div class="title unselectable">&debuggerUI.propertiesTitle;</div>
                 <div id="variables" class="vbox flex"></div>
             </div>
         </div>
-        <div id="dbg-statusbar">
-            <span id="status"></span>
-        </div>
     </div>
 
 </xul:window>
diff --git a/browser/themes/gnomestripe/devtools/common.css b/browser/themes/gnomestripe/devtools/common.css
--- a/browser/themes/gnomestripe/devtools/common.css
+++ b/browser/themes/gnomestripe/devtools/common.css
@@ -20,6 +20,7 @@
  *
  * Contributor(s):
  *   Paul Rouget <paul@mozilla.com> (original author)
+ *   Rob Campbell <rcampbell@mozilla.com>
  *
  * Alternatively, the contents of this file may be used under the terms of
  * either the GNU General Public License Version 2 or later (the "GPL"), or
@@ -35,6 +36,19 @@
  *
  * ***** END LICENSE BLOCK ***** */
 
+
+/* Splitters */
+
+.devtools-horizontal-splitter {
+  -moz-appearance: none;
+  border-top: 1px solid black;
+  border-bottom-width: 0;
+  min-height: 3px;
+  height: 3px;
+  margin-bottom: -3px;
+  position: relative;
+}
+
 /* Toolbar and Toolbar items */
 
 .devtools-toolbar {
@@ -44,6 +58,12 @@
   background: -moz-linear-gradient(top, hsl(210,11%,36%), hsl(210,11%,18%));
 }
 
+.devtools-toolbarlabel {
+  margin: 7px 6px;
+  color: hsl(210,30%,85%);
+  text-shadow: 0 -1px 0 hsla(210,8%,5%,.45);
+}
+
 .devtools-toolbarbutton {
   -moz-appearance: none;
   min-width: 78px;
diff --git a/browser/themes/gnomestripe/devtools/debugger.css b/browser/themes/gnomestripe/devtools/debugger.css
--- a/browser/themes/gnomestripe/devtools/debugger.css
+++ b/browser/themes/gnomestripe/devtools/debugger.css
@@ -24,6 +24,7 @@
  *   Dave Camp <dcamp@mozilla.com>
  *   Victor Porof <vporof@mozilla.com>
  *   Panos Astithas <past@mozilla.com>
+ *   Rob Campbell <rcampbell@mozilla.com>
  *
  * Alternatively, the contents of this file may be used under the terms of
  * either the GNU General Public License Version 2 or later (the "GPL"), or
@@ -50,44 +51,11 @@ a {
 }
 
 /**
- * Debugger toolbar
- */
-
-#dbg-toolbar {
-  padding: 2px;
-}
-
-#dbg-toolbar > button {
-  text-align: center;
-}
-
-/**
  * Debugger content
  */
 
 #dbg-content {
-  padding: 6px;
-}
-
-#dbg-content > * > .vbox {
-  background-color: #fff;
-  border: 1px solid #bbb;
-  box-shadow: 0 -5px 10px -3px #ccc;
-  -moz-margin-end: -1px;
-}
-
-#dbg-content > * > .title {
-  text-shadow: 0 1px #fff;
-}
-
-/**
- * Debugger statusbar
- */
-
-#dbg-statusbar {
-  font: -moz-list;
-  padding: 2px;
-  -moz-padding-start: 5px;
+  padding: 0;
 }
 
 /**
diff --git a/browser/themes/pinstripe/devtools/common.css b/browser/themes/pinstripe/devtools/common.css
--- a/browser/themes/pinstripe/devtools/common.css
+++ b/browser/themes/pinstripe/devtools/common.css
@@ -20,6 +20,7 @@
  *
  * Contributor(s):
  *   Paul Rouget <paul@mozilla.com> (original author)
+ *   Rob Campbell <rcampbell@mozilla.com>
  *
  * Alternatively, the contents of this file may be used under the terms of
  * either the GNU General Public License Version 2 or later (the "GPL"), or
@@ -37,6 +38,19 @@
 
 %include ../shared.inc
 
+
+/* Splitters */
+
+.devtools-horizontal-splitter {
+  -moz-appearance: none;
+  border-top: 1px solid black;
+  border-bottom-width: 0;
+  min-height: 3px;
+  height: 3px;
+  margin-bottom: -3px;
+  position: relative;
+}
+
 /* Toolbar and Toolbar items */
 
 .devtools-toolbar {
@@ -46,6 +60,12 @@
   background-image: url(background-noise-toolbar.png), -moz-linear-gradient(top, hsl(210,11%,36%), hsl(210,11%,18%));
 }
 
+.devtools-toolbarlabel {
+  margin: 7px 6px;
+  color: hsl(210,30%,85%);
+  text-shadow: 0 -1px 0 hsla(210,8%,5%,.45);
+}
+
 .devtools-toolbarbutton {
   -moz-appearance: none;
   min-width: 78px;
@@ -91,7 +111,7 @@
 
 .devtools-toolbarbutton[type=menu] > .toolbarbutton-menu-dropmarker,
 .devtools-toolbarbutton[type=menu-button] > .toolbarbutton-menubutton-dropmarker {
-  -moz-appearance: none!important;
+  -moz-appearance: none !important;
   list-style-image: url("chrome://browser/skin/devtools/dropmarker.png");
   -moz-box-align: center;
   margin: 0 3px;
diff --git a/browser/themes/pinstripe/devtools/debugger.css b/browser/themes/pinstripe/devtools/debugger.css
--- a/browser/themes/pinstripe/devtools/debugger.css
+++ b/browser/themes/pinstripe/devtools/debugger.css
@@ -24,6 +24,7 @@
  *   Dave Camp <dcamp@mozilla.com>
  *   Victor Porof <vporof@mozilla.com>
  *   Panos Astithas <past@mozilla.com>
+ *   Rob Campbell <rcampbell@mozilla.com>
  *
  * Alternatively, the contents of this file may be used under the terms of
  * either the GNU General Public License Version 2 or later (the "GPL"), or
@@ -50,48 +51,11 @@ a {
 }
 
 /**
- * Debugger toolbar
- */
-
-#dbg-toolbar {
-  padding: 2px;
-}
-
-#dbg-toolbar > button {
-  text-align: center;
-}
-
-#dbg-toolbar > button, menulist {
-  -moz-appearance: toolbarbutton;
-}
-
-/**
  * Debugger content
  */
 
 #dbg-content {
-  padding: 6px;
-}
-
-#dbg-content > * > .vbox {
-  background-color: #fff;
-  border: 1px solid #bbb;
-  box-shadow: 0 -5px 10px -3px #ccc;
-  -moz-margin-end: -1px;
-}
-
-#dbg-content > * > .title {
-  text-shadow: 0 1px #fff;
-}
-
-/**
- * Debugger statusbar
- */
-
-#dbg-statusbar {
-  font: -moz-list;
-  padding: 2px;
-  -moz-padding-start: 5px;
+  padding: 0;
 }
 
 /**
diff --git a/browser/themes/winstripe/devtools/common.css b/browser/themes/winstripe/devtools/common.css
--- a/browser/themes/winstripe/devtools/common.css
+++ b/browser/themes/winstripe/devtools/common.css
@@ -20,6 +20,7 @@
  *
  * Contributor(s):
  *   Paul Rouget <paul@mozilla.com> (original author)
+ *   Rob Campbell <rcampbell@mozilla.com>
  *
  * Alternatively, the contents of this file may be used under the terms of
  * either the GNU General Public License Version 2 or later (the "GPL"), or
@@ -35,6 +36,18 @@
  *
  * ***** END LICENSE BLOCK ***** */
 
+/* Splitters */
+
+.devtools-horizontal-splitter {
+  -moz-appearance: none;
+  border-top: 1px solid black;
+  border-bottom-width: 0;
+  min-height: 3px;
+  height: 3px;
+  margin-bottom: -3px;
+  position: relative;
+}
+
 /* Toolbar and Toolbar items */
 
 .devtools-toolbar {
@@ -44,6 +57,12 @@
   background-image: -moz-linear-gradient(top, hsl(209,18%,34%), hsl(210,24%,16%));
 }
 
+.devtools-toolbarlabel {
+  margin: 7px 6px;
+  color: hsl(210,30%,85%);
+  text-shadow: 0 -1px 0 hsla(210,8%,5%,.45);
+}
+
 .devtools-toolbarbutton {
   -moz-appearance: none;
   min-width: 78px;
diff --git a/browser/themes/winstripe/devtools/debugger.css b/browser/themes/winstripe/devtools/debugger.css
--- a/browser/themes/winstripe/devtools/debugger.css
+++ b/browser/themes/winstripe/devtools/debugger.css
@@ -24,6 +24,7 @@
  *   Dave Camp <dcamp@mozilla.com>
  *   Victor Porof <vporof@mozilla.com>
  *   Panos Astithas <past@mozilla.com>
+ *   Rob Campbell <rcampbell@mozilla.com>
  *
  * Alternatively, the contents of this file may be used under the terms of
  * either the GNU General Public License Version 2 or later (the "GPL"), or
@@ -50,44 +51,11 @@ a {
 }
 
 /**
- * Debugger toolbar
- */
-
-#dbg-toolbar {
-  padding: 2px;
-}
-
-#dbg-toolbar > button {
-  text-align: center;
-}
-
-/**
  * Debugger content
  */
 
 #dbg-content {
-  padding: 6px;
-}
-
-#dbg-content > * > .vbox {
-  background-color: #fff;
-  border: 1px solid #bbb;
-  box-shadow: 0 -5px 10px -3px #ccc;
-  -moz-margin-end: -1px;
-}
-
-#dbg-content > * > .title {
-  text-shadow: 0 1px #fff;
-}
-
-/**
- * Debugger statusbar
- */
-
-#dbg-statusbar {
-  font: -moz-list;
-  padding: 2px;
-  -moz-padding-start: 5px;
+  padding: 0;
 }
 
 /**

